app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@stanhu do you think the following way to check the file size is okay? The user is redirect to the project repository tree if the file size exceeds 10MB, and the reminding information is also shown.
![file_size](https://cloud.githubusercontent.com/assets/7370334/9134762/b5e0283e-3d07-11e5-8ec2-0e89220bd241.png)
![reminder](https://cloud.githubusercontent.com/assets/7370334/9134763/b5e17a86-3d07-11e5-9d4c-056d3839cae1.png)